pkg/lib/interaction/nodes/create_authenticator_oob_setup.go

```diff
@@ -55,6 +57,26 @@ func (e *EdgeCreateAuthenticatorOOBSetup) Instantiate(ctx *interaction.Context,
 		target = input.GetOOBTarget()
 	}
 
+	// Validate target against channel
+	validationCtx := &validation.Context{}
+	switch channel {
+	case authn.AuthenticatorOOBChannelEmail:
+		err := validation.FormatEmail{AllowName: false}.CheckFormat(target)
+		if err != nil {
+			validationCtx.EmitError("format", map[string]interface{}{"format": "email"})
+		}
+	case authn.AuthenticatorOOBChannelSMS:
+		err := validation.FormatPhone{}.CheckFormat(target)
+		if err != nil {
+			validationCtx.EmitError("format", map[string]interface{}{"format": "phone"})
+		}
+	}
+
+	err := validationCtx.Error("invalid target")
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
The original logic in handler not only validate the phone number, it also do parse. That means the phone number we store in DB is a tidied up format (e.g. no space and hyphen..etc). Maybe we should
1. check if phone login id parsed inside interaction?
2. ensure oob also parse the phone

Reply:
I double checked and the current behavior is that the raw input from user must be in E.164 format, otherwise a validation error will be thrown.